### PR TITLE
CDPS-980: Add json history field for medical diet/allergies

### PR DESCRIPTION
**Pull request by danielburnley**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt [hunk 1]
```diff
@@ -34,6 +35,9 @@ class FieldHistory(
   override var valueInt: Int? = null,
   override var valueString: String? = null,
 
+  @Convert(converter = JsonObjectConverter::class)
+  override var valueJson: JsonObject? = null,
```
Comment (davidatmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt:39]: Need to include `valueJson` in `equals`, `hashCode` and `toString`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/enums/PrisonPersonField.kt [hunk 2]
```diff
@@ -1,29 +1,68 @@
...
+  WEIGHT(getInt, setInt, hasChangedInt, "WEIGHT"),
+  HAIR(getRef, setRef, hasChangedRef, "HAIR"),
+  FACIAL_HAIR(getRef, setRef, hasChangedRef, "FACIAL_HAIR"),
+  FACE(getRef, setRef, hasChangedRef, "FACE"),
+  BUILD(getRef, setRef, hasChangedRef, "BUILD"),
+  LEFT_EYE_COLOUR(getRef, setRef, hasChangedRef, "EYE"),
+  RIGHT_EYE_COLOUR(getRef, setRef, hasChangedRef, "EYE"),
+  SHOE_SIZE(getString, setString, hasChangedString, null),
+  SMOKER_OR_VAPER(getRef, setRef, hasChangedRef, "SMOKE"),
+  FOOD_ALLERGY(
```
Comment (davidatmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/enums/PrisonPersonField.kt:38]: Could extract into `getJson`, `setJson`, `hasChangedJson` to align with Int, String and Ref

Comment (brightonsbox) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/enums/PrisonPersonField.kt:38]: Might get a bit tricky to make generic given the need to convert to concrete classes?

Comment (danielburnley) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/enums/PrisonPersonField.kt:38]: I can probably extract the `getJson` but the subsequent 2 are going to be unique to each json field

**Review by davidatmoj — APPROVED** [review]

**Pull request merged**